app/src/main/java/no/usn/mob3000/data/repository/social/ProfileEditRepository.kt

```diff
@@ -0,0 +1,57 @@
+package no.usn.mob3000.data.repository.social
+
+import kotlinx.datetime.Clock
+import no.usn.mob3000.data.model.social.ProfileDto
+import no.usn.mob3000.data.source.remote.auth.UserDataSource
+import no.usn.mob3000.data.source.remote.social.ProfileUserDataSource
+import no.usn.mob3000.domain.repository.social.IProfileEditRepository
+/**
+ * This repository orchestrates the updating of the users profile. Currently there is only one implementation of
+ * updating a table under profile, thus [updateProfile] being the sole function here
+ *
+ * @param profileEditDataSource The data source for fetching friends.
+ * @author 258030
+ * @created 2024-11-15
+ */
+class ProfileEditRepository (
+    private val profileEditDataSource: ProfileUserDataSource = ProfileUserDataSource(),
+    private val userDataSource: UserDataSource = UserDataSource()
+):IProfileEditRepository{
+    /**
+     *  Updates the users profile. As long as the value of the user exist, we try to update. Else throw exception
+     *
+     *  @param userid The user id of the current user.
+     *  @param displayName The display name of the user.
+     *  @param avatarUrl The avatar url of the user.
+     *  @param aboutMe The about me of the user.
+     *  @param nationality The nationality of the user.
+     *  @param profileVisibility The profile visibility of the user.
+     *  @param friendsVisibility The friends visibility of the user.
+     */
+    override suspend fun updateProfile(
+        userid: String,
+        displayName: String,
+        avatarUrl: String,
+        aboutMe: String,
+        nationality: String,
+        profileVisibility: Boolean,
+        friendsVisibility: Boolean
+    ) : Result<Unit>{
```

Comment:
Does not extend the result. Incorrect.

Reply:
Does the code work as intented? Are you saying this functionality does not do what it promises? This works the excact same way:
`override suspend fun updateNews(
        newsId: String,
        title: String,
        summary: String,
        content: String,
        isPublished: Boolean
    ): Result<Unit> {
        val originalNews = newsDataSource.fetchNewsById(newsId)
        if (originalNews != null) {
            val updatedNewsDto = NewsDto(
                newsId = newsId,
                createdAt = originalNews.createdAt,
                modifiedAt = Clock.System.now(),
                createdByUser = originalNews.createdByUser,
                title = title,
                summary = summary,
                content = content,
                isPublished = isPublished
            )
            return newsDataSource.updateNews(newsId, updatedNewsDto)
        } else {
            return Result.failure(Exception("Original news data not found"))
        }
    }`

Reply:
It is not about the actual result. It's about the line this comment is placed on. `Result<Unit>` is what the function returns to callers of the function, however, it does not extend or implement it. And it's also not a class, for that matter. It should be `): Result<Unit>`. Additionally, spacing between the return value and the opening bracket needs to be added too. `t> {`.

This is an Android Developers style guide thing. By Kotlin's style guide, this would be correct placement of a return value as far as I understand.